src/main/java/mate/academy/dao/impl/UserDaoImpl.java

```diff
@@ -0,0 +1,58 @@
+package mate.academy.dao.impl;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import mate.academy.dao.UserDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.User;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+@Dao
+public class UserDaoImpl implements UserDao {
+    @Override
+    public User add(User user) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.persist(user);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can`t add user to DB " + user, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return user;
+    }
+
+    @Override
+    public Set<String> getAllLogins() {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            List<String> logins = session.createQuery("SELECT login FROM User", String.class)
+                    .getResultList();
+            return new HashSet<>(logins);
+        } catch (Exception e) {
+            throw new DataProcessingException("Can`t get all users` logins.", e);
+        }
+    }

```

Comment:
Its a good point to check if a login exists, but imagine you have millions of users, selecting all logins not the best idea in that case. Try to use this query "SELECT COUNT(*) FROM users WHERE login = ?" and if count > 0 - it means that you already had a user with such login